src/formatters/Utils.cc

```diff
@@ -286,6 +286,24 @@ std::string formatFitsProperties(daf::base::PropertyList const& prop,
 int countFitsHeaderCards(lsst::daf::base::PropertySet const& prop) {
     return prop.paramNames(false).size();
 }
+
+ndarray::Array<std::uint8_t, 1, 1> stringToBytes(std::string const& str) {
+    auto nbytes = str.size() * sizeof(char) / sizeof(std::uint8_t);
```

Comment:
`sizeof(typename std::string::value_type)`? Or do we trust the user to update this if they switch the input to some other string type?

Reply:
In my code if somebody changes the argument type then the code will continue to operate correctly (though I hate to think what will happen when reading old persisted data). In your suggested change the type appears in two different places which increases the risk that they will go out of synch if somebody changes the type of the `str` argument. So I think my code is safer than yours. What am I missing?

Reply:
I don't see why your suggested change is any better -- my code and yours both seem to need updating if a user changes the type of the `str` argument. For the length computation to be safe if the type of `str` changes perhaps we want `sizeof(str[0])`?

Of course if the type of the `str` argument changes then old persisted data will probably not unpersist correctly.

Reply:
`std::string::data` is documented to return `const char *` so I think the current code is safest.